nemo_rl/models/policy/megatron_policy_worker.py

```diff
@@ -1559,21 +1665,15 @@ def get_weights_ipc_handles(self, *, keys: list[str]) -> dict[str, Any]:
     @torch.no_grad()
     def broadcast_weights_for_collective(self) -> None:
         """Broadcast the weights for collective communication."""
-        for key, _ in self.refit_param_info_mcore:
-            # gather megatron params
-            gathered_megatron_params = gather_params(
-                self.model,
-                [key],
-                key_to_global_keys=self.local_key_to_global_keys,
-            )
-            # convert to hf params
-            gathered_hf_params = self.megatron_to_hf_converter.convert(
-                gathered_megatron_params, self.model.config
-            )
-            # broadcast from train rank0 worker to inference workers
-            if self.rank == 0:
-                for _, tensor in gathered_hf_params.items():
-                    self.model_update_group.broadcast(tensor, src=0)
+        hf_params_generator = self.megatron_bridge.export_hf_weights(
+            [self.model],
+            show_progress=False,
+        )
+        gathered_hf_params = {name: tensor for name, tensor in hf_params_generator}
+        # broadcast from train rank0 worker to inference workers
+        if self.rank == 0:
+            for _, tensor in gathered_hf_params.items():
+                self.model_update_group.broadcast(tensor, src=0)
```

Comment:
feels current implementation will OOM at `gathered_hf_params = {name: tensor for name, tensor in hf_params_generator}`, since it will gather all the params of the model.

do you think it's ok to change to the below?

```suggestion
        hf_params_generator = self.megatron_bridge.export_hf_weights(
            [self.model],
            show_progress=False,
        )
        # broadcast from train rank0 worker to inference workers
        for _, tensor in hf_params_generator:
            if self.rank == 0:
                self.model_update_group.broadcast(tensor, src=0)
```